services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/allu/CableReportServiceAllu.kt

```diff
@@ -204,18 +205,28 @@ class CableReportServiceAllu(
             .orElseThrow()
     }
 
-    override fun getDecisionPDF(applicationId: Int): ByteArray {
+    override fun getDecisionPdf(applicationId: Int): ByteArray {
         val token = login()!!
         return webClient
             .get()
             .uri("$baseUrl/v2/cablereports/$applicationId/decision", applicationId)
             .accept(MediaType.APPLICATION_PDF)
             .headers { it.setBearerAuth(token) }
-            .exchange()
+            .retrieve()
+            .onStatus(
+                { httpStatus -> httpStatus.value() == 404 },
+                {
+                    Mono.error(
+                        ApplicationDecisionNotFoundException(
+                            "Decision not found in Allu. alluid=$applicationId"
+                        )
+                    )
+                }
+            )
+            .bodyToMono(ByteArrayResource::class.java)
```

Comment:
Can we be sure that the request will indeed return a pdf? Would a check for content type in the response headers make any sense?

Reply:
Allu API description says it should respond with `application/pdf`. https://testi.allu.kaupunkiymparisto.fi/external/swagger-ui/index.html?urls.primaryName=v2#/Cable%20reports/getDecision_5

We tell Allu that we `Accept: application/pdf`, so it shouldn't send us anything else, at least if the response is Ok.

Then again, checking the headers doesn't really hurt us, so I added it.